packages/catalog-server/src/lib/firestore/package-info-converter.ts

```diff
@@ -17,26 +17,37 @@ import {
   isReadablePackage,
   PackageInfo,
   ReadablePackageInfo,
+  ReadablePackageStatus,
+  UnreadablePackageInfo,
 } from '@webcomponents/catalog-api/lib/schema.js';
+import { distTagListToMap } from '../npm.js';
 
-export const packageInfoConverter: FirestoreDataConverter<
-  Omit<PackageInfo, 'version'>
-> = {
-  fromFirestore(
-    snapshot: QueryDocumentSnapshot<DocumentData>
-  ): Omit<PackageInfo, 'version'> {
-    const distTags = snapshot.get('distTags');
-    const graphQLDistTags = Object.entries(distTags).map(
-      ([tag, version]) => ({tag, version} as DistTag)
-    );
-    return {
-      name: idToPackageName(snapshot.id),
-      lastUpdate: (snapshot.get('lastUpdate') as Timestamp).toDate(),
-      status: snapshot.get('status'),
-      description: snapshot.get('description'),
-      distTags: graphQLDistTags,
-      // `version` is left to a sub-collection query
-    } as ReadablePackageInfo;
+export const packageInfoConverter: FirestoreDataConverter<PackageInfo> = {
+  fromFirestore(snapshot: QueryDocumentSnapshot<DocumentData>): PackageInfo {
+    const name = idToPackageName(snapshot.id);
+    const status = snapshot.get('status');
+    const lastUpdate = (snapshot.get('lastUpdate') as Timestamp).toDate();
+
+    if (status === ReadablePackageStatus.READY || status === ReadablePackageStatus.UPDATING) {
```

Comment:
Use the `isReadablePackage` utility?

Reply:
This is not a PackageInfo object, it's a Firestore DocumentSnapshot.